lib/src/query_component.dart

```diff
@@ -72,25 +116,34 @@ class QueryComponent {
     final jsonCreateQueryResponse = await _queryService
         .execReportingCreateQuery(advertiserId, insertionOrderId, dateRange);
     final reportingQueryId = ReportingQueryParser.parseQueryIdFromJsonString(
-        json.stringify(jsonCreateQueryResponse));
+        JsonJS.stringify(jsonCreateQueryResponse));
 
     try {
       // Uses the queryId to get the report download path.
       final jsonGetQueryResponse =
-      await _queryService.execReportingGetQuery(reportingQueryId);
+          await _queryService.execReportingGetQuery(reportingQueryId);
       final reportingDownloadPath =
-      ReportingQueryParser.parseDownloadPathFromJsonString(
-          json.stringify(jsonGetQueryResponse));
+          ReportingQueryParser.parseDownloadPathFromJsonString(
+              JsonJS.stringify(jsonGetQueryResponse));
 
       // Downloads the report and parse the response into a revenue map.
       final report =
-      await _queryService.execReportingDownload(reportingDownloadPath);
+          await _queryService.execReportingDownload(reportingDownloadPath);
 
       return ReportingQueryParser.parseRevenueFromJsonString(report);
-    } catch(e) {
+    } catch (e) {
       /// TODO: proper error handling.
       /// Issue: https://github.com/googleinterns/dv360-excel-plugin/issues/52.
       return <String, String>{};
     }
   }
+
+  List<InsertionOrder> _filterInFlightInsertionOrders(
+      List<InsertionOrder> insertionOrders) {
+    return insertionOrders
+        .where((io) =>
+            io.budget.activeBudgetSegment !=
+            InsertionOrder_Budget_BudgetSegment())
```

Comment:
this doesnt need to check dates?

Reply:
If the IO doesn't have an activeBudgetSegment, it will have an empty InsertionOrder_Budget_BudgetSegment(). 

Date checking has already been done during [parsing](https://github.com/googleinterns/dv360-excel-plugin/blob/4b50b34a1e2309e47deb360aa25076d4353a82cf/lib/src/dv360_service_response_parser.dart#L115-L118).